add ImmediateContainer, Quantity and SimplyUnit, clarify Unit use (#1689)
open-contracting/standard: docs/schema/reference.md

#### Unit

> The `unit` block allows detailed specification of the parameters and price of units that make up a line-item. It should be used for units of measurement only. For units of presentation it is recommended to use the `ImmediateContainer` subschema.

Reviewer: We probably ought to define unit of measurement for clarity.

```suggestion
The `unit` subschema allows detailed specification of the parameters and price of units that make up a line-item. It should be used for units of measurement only. For units of presentation, use the [`ImmediateContainer` subschema](#immediatecontainer), which is used by the [`Item.immediateContainer` field](#item).
```

Author: have added in a definition here of what a unit of measurement is as well as accepting this suggestion

:collapse: value
```

#### ImmediateContainer

> The `ImmediateContainer` subschema allows detailed specification of the immediate container of an item. This is also known as the presentation, the presentation form, the package. For units of measurement it is recommended to use the `Unit` subschema.

Reviewer: ```suggestion
The `ImmediateContainer` subschema allows detailed specification of the immediate container of an item. This is also called the presentation, presentation form, or package. For units of measurement, use the [`Unit` subschema](#unit).
```
